rs/embedders/src/signal_handler.rs

```diff
@@ -40,6 +40,10 @@ pub(crate) fn sigsegv_memory_tracker_handler(
     move |signum: i32, siginfo_ptr: *const libc::siginfo_t, ucontext_ptr: *const libc::c_void| {
         use nix::sys::signal::Signal;
 
+        // The timer starts immediately. The guard will capture this timer and report it
+        // to the corresponding metric when the `memory_tracker` object is locked.
```

Comment:
Did you maybe mean "dropped" instead of "locked"?

Reply:
We start the timer here, but we can't create a guard until the `memory_tracker` is locked below in this line: `let memory_tracker = memory_tracker.lock().unwrap();`

Reply:
I rephrased the comment...